src/robot.py

```diff
@@ -213,7 +232,13 @@ def update_end_pose(self, end_pose):
         self.end_pose = end_pose
 
     def is_done(self):
-        return self.current_pose == self.end_pose
+        return self.current_pose == self.charging_station and self.has_shelf
+    
+    def reset(self, end_pose):
```

Comment:
If you want to keep them separate, then this is the perfect function to keep.  I'd rename it to something like `set_new_endpoint`, because it's more of that and less of a reset.